src/ConnectedMode/Binding/ImportBeforeFileGenerator.cs

```diff
@@ -21,77 +21,79 @@
+        private void WriteTargetsFileToDiskIfNotExists()
         {
-            var sb = new StringBuilder();
-            var xmlWriter = Create(sb);
+            logger.LogVerbose(Resources.ImportBeforeFileGenerator_CheckingIfFileExists);
 
-            xmlWriter.WriteStartElement("Project", "http://schemas.microsoft.com/developer/msbuild/2003");
-            xmlWriter.WriteAttributeString("xmlns", "http://schemas.microsoft.com/developer/msbuild/2003");
-
-            xmlWriter.WriteStartElement("ItemGroup");
-
-            xmlWriter.WriteEndElement();
-            xmlWriter.WriteEndElement();
-
-            xmlWriter.Close();
-            WriteFile(sb.ToString());
-        }
-
-        private void WriteFile(string xml)
-        {
+            var resource = ReadResourceFile();
             var pathToImportBefore = GetPathToImportBefore();
+            var fullPath = Path.Combine(pathToImportBefore, targetsFileName);
 
-            if (!fileSystem.Directory.Exists(pathToImportBefore))
+            try
             {
-                fileSystem.Directory.CreateDirectory(pathToImportBefore);
+                if (!fileSystem.Directory.Exists(pathToImportBefore))
+                {
+                    logger.LogVerbose(Resources.ImportBeforeFileGenerator_CreatingDirectory, pathToImportBefore);
+                    fileSystem.Directory.CreateDirectory(pathToImportBefore);
+                }
+
+                if (fileSystem.File.Exists(fullPath) && fileSystem.File.ReadAllText(fullPath) == resource)
+                {
+                    logger.LogVerbose(Resources.ImportBeforeFileGenerator_FileAlreadyExists);
+                    return;
+                }
+
+                logger.LogVerbose(Resources.ImportBeforeFileGenerator_WritingTargetFileToDisk);
+                fileSystem.File.WriteAllText(fullPath, resource);
             }
-
-            var fullPath = Path.Combine(pathToImportBefore, targetsFileName);
-
-            if (fileSystem.File.Exists(fullPath) && fileSystem.File.ReadAllText(fullPath) == xml)
+            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
             {
-                return;
+                logger.WriteLine(Resources.ImportBeforeFileGenerator_FailedToWriteFile);
+                logger.LogVerbose(Resources.ImportBeforeFileGenerator_FailedToWriteFile_Verbose, ex.Message);
             }
-
-            fileSystem.File.WriteAllText(fullPath, xml);
         }
 
-        private XmlWriter Create(StringBuilder sb)
+        private string ReadResourceFile()
         {
-            var stringWriter = new StringWriter(sb);
+            logger.LogVerbose(Resources.ImportBeforeFileGenerator_LoadingResourceFile);
 
-            var settings = new XmlWriterSettings
+            var resourcePath = "SonarLint.VisualStudio.ConnectedMode.Embedded.SonarLintTargets.txt";
+
+            using (var stream = GetType().Assembly.GetManifestResourceStream(resourcePath))
```

Comment:
This could just be 

``
var resourcePath = "SonarLint.VisualStudio.ConnectedMode.Embedded.SonarLintTargets.txt";
using var stream = new 
                       StreamReader(typeof(ImportBeforeFileGenerator).Assembly.GetManifestResourceStream(resourcePath));

return stream.ReadToEnd();
```

instead of all this but I saw it done like this in a different file and as it is safe I suppose this makes more sense